From 1538acbcc0e4d64abcb1475c3f6df5ba9df9caa6 Mon Sep 17 00:00:00 2001
From: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>
Date: Thu, 15 Jun 2017 15:37:34 -0400
Subject: [PATCH 2090/2094] drm/amdgpu: Fix address mask in get_local_mem_info

Use ASIC specific dma_mask instead of hard coding address mask.

Change-Id: I2bbb54c685cefe05ca48794d0e6b3229c313db99
Signed-off-by: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
index 4a48483..bbd8afa 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd.c
@@ -317,7 +317,8 @@ void get_local_mem_info(struct kgd_dev *kgd,
 	resource_size_t aper_limit;
 	struct amdgpu_device *rdev = (struct amdgpu_device *)kgd;
 
-	address_mask = ~((1UL << 40) - 1);
+	address_mask = rdev->dev->dma_mask ? ~*rdev->dev->dma_mask :
+					     ~((1ULL << 32) - 1);
 	aper_limit = rdev->mc.aper_base + rdev->mc.aper_size;
 	memset(mem_info, 0, sizeof(*mem_info));
 	if (!(rdev->mc.aper_base & address_mask ||
-- 
2.7.4

